Knossos.NET/ViewModels/Windows/ModInstallViewModel.cs

```diff
@@ -258,7 +260,23 @@ private async Task ProcessMod(Mod mod, List<Mod> allMods, List<Mod>? processed =
                                 }
                             }
                         }
+
+                        //If mod is already installed change non-installed, recommended packages to optional
+                        //and all installed ones to requiered since we dont support uninstalling mod pkgs
+                        if (pkg.status != "required" && modInstalled != null && modInstalled.packages != null)
+                        {
+                            var packageIsInstalled = modInstalled.packages.FirstOrDefault(m => m.name == pkg.name);
+                            if (packageIsInstalled == null)
+                            {
+                                pkg.status = "optional";
+                            }
+                            else
+                            {
+                                pkg.status = "required";
+                            }
```

Comment:
What about, was previously installed, but originally optional?  Does that now become required?

Reply:
Good question. Everything thats already installed may become requiered during modify... mmmm and during install too if the originally optinal package is requiered by other mod. Since we dont support unistalling pkgs, at least for now, this was not important but i dont like it. Ill to check if that is indeed the case.